### DISCUSSION TITLE
Terrarium Project Part 3

### DISCUSSION THREAD

**Discussion by bodV6** [microsoft/Web-Dev-For-Beginners#710] [Q&A]
uncaught reference error: terrariumElement not defined

terrariumElement parameter in dragElement closure function declaration is not available to use as argument in elementDrag```
function which takes event as para...

kindly help make the App work.

**Comment by ShubhamMaster** [comment]
The error simply means: terrariumElement is being used in a place where it doesn’t exist.
Most likely your structure looks like this:
JavaScript
function dragElement(terrariumElement) {
}

function elementDrag(e) {
    terrariumElement.style.top = ... // error
}
Here, terrariumElement is only defined inside dragElement, but elementDrag is outside, so it cannot access it.
Fix 1 (best approach)
Keep elementDrag inside dragElement so it shares the same scope:
JavaScript
function dragElement(terrariumElement) {

    terrariumElement.onmousedown = dragMouseDown;

    function dragMouseDown(e) {
        document.onmousemove = elementDrag;
    }

    function elementDrag(e) {
        terrariumElement.style.top = "100px";
    }
}
Fix 2 (if you want separate functions)
Pass the element explicitly:
JavaScript
function dragElement(terrariumElement) {
    document.onmousemove = (e) => elementDrag(e, terrariumElement);
}

function elementDrag(e, terrariumElement) {
    terrariumElement.style.top = "100px";
}
Key point
terrariumElement is not global.
It only exists where you pass it. If another function needs it, you must:
either keep that function inside, or
pass it again as a parameter.
